src/utils.tsx

```diff
@@ -94,3 +95,22 @@ export function uuidV4 ():string {
     return value.toString(16)
   })
 }
+
+export function useNonInitialEffect (effect: EffectCallback, deps?: DependencyList): void {
+  const initialRender = useRef(true)
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    let effectReturns: void | (() => void) = () => {}
+
+    if (initialRender.current) {
+      initialRender.current = false
+    } else {
+      effectReturns = effect()
+    }
+
+    if (effectReturns && typeof effectReturns === 'function') {
```

Comment:
You could have only checked the type, no ? if its a function, it is truthy by default I think

Reply:
Good